build.gradle

```diff
@@ -325,48 +324,63 @@ signing {
-
-            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
-                authentication(userName: project.findProperty("sonatypeUsername"), password: project.findProperty("sonatypePassword"))
-            }
-
-            snapshotRepository(url: "https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/") {
-                authentication(userName: System.env.ARTIFACTORY_USERNAME, password: System.env.ARTIFACTORY_PASSWORD)
-            }
-
-            pom.project {
-                name 'Picard'
-                packaging 'jar'
-                description 'A set of command line tools (in Java) for manipulating high-throughput sequencing (HTS) data and formats such as SAM/BAM/CRAM and VCF.'
-                url 'http://broadinstitute.github.io/picard/'
-
-                developers {
-                    developer {
-                        id 'picard'
-                        name 'Picard Team'
-                        url 'http://broadinstitute.github.io/picard'
-                    }
-                }
+publishing {
+    publications {
+        picard(MavenPublication) {
+            from components.java
+
+            artifact sourcesJar
+            artifact javadocJar
+            pom {
+                name = 'Picard'
+                packaging = 'jar'
+                description = 'A set of command line tools (in Java) for manipulating high-throughput sequencing (HTS) data and formats such as SAM/BAM/CRAM and VCF.'
+                url = 'http://broadinstitute.github.io/picard/'
 
                 scm {
-                    url 'git@github.com:broadinstitute/picard.git'
-                    connection 'scm:git:git@github.com:broadinstitute/picard.git'
+                    url = 'git@github.com:broadinstitute/picard.git'
+                    connection = 'scm:git:git@github.com:broadinstitute/picard.git'
                 }
 
                 licenses {
                     license {
-                        name 'MIT License'
-                        url 'http://opensource.org/licenses/MIT'
-                        distribution 'repo'
+                        name = 'MIT License'
+                        url = 'http://opensource.org/licenses/MIT'
+                        distribution = 'repo'
+                    }
+                }
+
+                developers {
+                    developer {
+                        id = 'picard'
+                        name = 'Picard Team'
+                        url = 'http://broadinstitute.github.io/picard'
                     }
                 }
             }
         }
     }
+    repositories {
+        maven {
+            url = "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+            credentials {
+                username = project.findProperty("sonatypeUsername")
+                password = project.findProperty("sonatypePassword")
+            }
+        }
+        maven {
```

Comment:
Does this work?   In gatk / htsjdk we do this:

```

    repositories {
        maven {
            credentials {
                username = isRelease ? project.findProperty("sonatypeUsername") : System.env.ARTIFACTORY_USERNAME
                password = isRelease ? project.findProperty("sonatypePassword") : System.env.ARTIFACTORY_PASSWORD
            }
            def release = "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
            def snapshot = "https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/"
            url = isRelease ? release : snapshot
        }
    }
```

Reply:
I'll give it a test and see what happens.